nameko/messaging.py

```diff
@@ -449,12 +228,26 @@ def handle_message(self, body, message):
         context_data = self.unpack_message_headers(message)
 
         handle_result = partial(self.handle_result, message)
-        try:
-            self.container.spawn_worker(self, args, kwargs,
-                                        context_data=context_data,
-                                        handle_result=handle_result)
-        except ContainerBeingKilled:
-            self.queue_consumer.requeue_message(message)
+
+        def spawn_worker():
+            try:
+                self.container.spawn_worker(
+                    self, args, kwargs,
+                    context_data=context_data,
+                    handle_result=handle_result
+                )
+            except ContainerBeingKilled:
+                self.consumer.requeue_message(message)
+
+        service_name = self.container.service_name
+        method_name = self.method_name
+
+        # TODO replace global worker pool limits with per-entrypoint limits,
+        # then remove this waiter thread
+        ident = u"{}.wait_for_worker_pool[{}.{}]".format(
+            type(self).__name__, service_name, method_name
+        )
+        self.container.spawn_managed_thread(spawn_worker, identifier=ident)
```

Comment:
This workaround was present in the QueueConsumer and must now be implemented in every entrypoint that consumes AMQP messages